internal/controllers/gateway/gateway_controller.go

```diff
@@ -54,6 +55,9 @@ type GatewayReconciler struct { //nolint:revive
 	PublishServiceRef    k8stypes.NamespacedName
 	PublishServiceUDPRef mo.Option[k8stypes.NamespacedName]
 
+	AddressOverrides    []string
+	AddressOverridesUDP []string
```

Comment:
Can we add comments to those field so that a reader knows exactly what this will be used for?